### PR TITLE
printer: print commit hash version for extensions

### LINKED ISSUE #43096: Show version info for the audit log in `tidb_version`
It is better that we can see the exact audit log version in tidb enterprise by `select tidb_version()`.

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What problem does this PR solve?

Issue Number: close #43096

Problem Summary:

### What is changed and how it works?

Add a new item for the printer to print the potential commit hash value of the enterprise extension.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

```
shell> make server

tidb > select tidb_version() \G
...
Store: unistore
...

shell> make enterprise-server
...-X "github.com/pingcap/tidb/util/versioninfo.TiDBEnterpriseExtensionVersion=216228961c416197dbd92e12ba5cedf5d64cf5eb"...

tidb> select tidb_version() \G
...
Store: unistore
Enterprise Extension Commit Hash: 216228961c416197dbd92e12ba5cedf5d64cf5eb
...
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Add a new item in tidb info to show the version of the enterprise extension.
```

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -41,6 +41,12 @@ ifneq "$(TIDB_EDITION)" "Enterprise"
 endif
 endif
 
+ENTERPRISE_EXTENSION_VERSION :=
```
Comment (lcwangchao) [inline Makefile.common:44]: I think it is `ENTERPRISE_EXTENSION_GIT_HASH` in fact ?

Comment (CbcWestwolf) [reply Makefile.common:44]: Yes, `ENTERPRISE_EXTENSION_GIT_HASH` is more exact. Let me correct it.

File: Makefile.common [hunk 2]
```diff
@@ -66,6 +66,10 @@ LDFLAGS += -X "github.com/pingcap/tidb/util/versioninfo.TiDBGitHash=$(shell git
 LDFLAGS += -X "github.com/pingcap/tidb/util/versioninfo.TiDBGitBranch=$(shell git rev-parse --abbrev-ref HEAD)"
 LDFLAGS += -X "github.com/pingcap/tidb/util/versioninfo.TiDBEdition=$(TIDB_EDITION)"
 
+ifeq ($(shell if [ -a extension/enterprise/.git ]; then echo "true"; fi),true)
```
[the remark this answers is not in the document]
Comment (CbcWestwolf) [reply Makefile.common:70]: Resolved in https://github.com/pingcap/tidb/pull/43095/commits/973a0a481fd2d368abeacb2edbb9bdbba25aef19

File: util/printer/printer.go [hunk 3]
```diff
@@ -59,7 +68,8 @@ func GetTiDBInfo() string {
 		"Race Enabled: %v\n"+
 		"TiKV Min Version: %s\n"+
 		"Check Table Before Drop: %v\n"+
-		"Store: %s",
+		"Store: %s"+
```
Comment (xhebox) [inline util/printer/printer.go:71]: Maybe add `\n` here.

File: Makefile.common (hunk 2 above)
[the remark this answers is not in the document]
Comment (xhebox) [reply Makefile.common:70]: This could be added into `LDFLAGS` directly.

**Review by xhebox — APPROVED** [review]

File: Makefile.common [hunk 4]
```diff
@@ -66,6 +66,11 @@ LDFLAGS += -X "github.com/pingcap/tidb/util/versioninfo.TiDBGitHash=$(shell git
 LDFLAGS += -X "github.com/pingcap/tidb/util/versioninfo.TiDBGitBranch=$(shell git rev-parse --abbrev-ref HEAD)"
 LDFLAGS += -X "github.com/pingcap/tidb/util/versioninfo.TiDBEdition=$(TIDB_EDITION)"
 
+EXTENSION_FLAG =
+ifeq ($(shell if [ -a extension/enterprise/.git ]; then echo "true"; fi),true)
+	EXTENSION_FLAG += -X "github.com/pingcap/tidb/util/versioninfo.TiDBEnterpriseExtensionVersion=$(shell cd extension/enterprise && git rev-parse HEAD)"
```
Comment (hawkingrei) [inline Makefile.common:71]: Bazel also need to add this info at ```build/print-workspace-status.sh```.

Comment (hawkingrei) [reply Makefile.common:71]: BTW if it is necessary, we can create ```build/print-workspace-status-with-plugin.sh```.

Comment (CbcWestwolf) [reply Makefile.common:71]: Add in https://github.com/pingcap/tidb/pull/43095/commits/6a5fc14050e05359e63da85c02d6a8784a77855a

**Review by lcwangchao — APPROVED** [review]

**Pull request merged**